client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -179,16 +162,17 @@ public String getContentType() {
      * @return The status code.
      */
     public int getStatus() {
-        return status;
+        return MessageHelper.getApplicationProperty(message.getApplicationProperties(),
+                MessageHelper.APP_PROPERTY_STATUS, Integer.class);
     }
 
     /**
-     * Gets a map of the properties to include in the message as application properties.
+     * Gets the AMQP 1.0 message representing this command response.
      *
-     * @return The properties.
+     * @return The command response message.
      */
-    public Map<String, Object> getProperties() {
-        return properties;
+    public Message getCommandResponseMessage() {
```

Comment:
IMHO something like `toMessage()` would better reflect the purpose of transforming this object to an AMQP message. The client doesn't know (and doesn't need to know) that, in fact, we keep an instance of `Message` internally all the time ...

Reply:
In the command class, it is named as `getCommandMessage` and I thought of keeping the same naming convention. I agree with `toMessage()`.